### PR TITLE
MAT-7450 API to retrieve Parameters, Definitions, Functions, and Fluent Functions from CQL

### PR DESCRIPTION
**Pull request by adongare**
## CQL to ELM Translation Service PR

Jira Ticket: [MAT-7450](https://jira.cms.gov/browse/MAT-7450)
(Optional) Related Tickets:

### Summary
- MAT-7450 API to retrieve Parameters, Definitions, Functions, and Fluent Functions from CQL
- Removed QDM lookup related code
- Stop visiting/listening on same library multiple times unnecessary

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlConversionController.java [hunk 1]
```diff
@@ -63,7 +65,7 @@ public CqlConversionPayload cqlToElmJson(
             .validateUnits(validateUnits)
             .resultTypes(resultTypes)
             .build();
-
+    cqlParsingService.getCqlBuilderLookups(cqlData, accessToken);
```
Comment (sb-cecilialiu) [inline src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlConversionController.java]: I'm wondering what is the use of getCqlBuilderLookups here?

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlToolsController.java [hunk 2]
```diff
@@ -92,15 +91,11 @@ public ResponseEntity<Map<String, Set<CQLDefinition>>> getDefinitionCallstack(
...
       produces = MediaType.APPLICATION_JSON_VALUE,
       consumes = MediaType.APPLICATION_JSON_VALUE)
-  public ResponseEntity<CqlLookups> getCqlLookups(
-      @RequestBody CqlLookupRequest lookupRequest,
-      @RequestHeader("Authorization") String accessToken) {
-    log.info("Translator is preparing CQL Lookups for simple xml");
-    return ResponseEntity.ok(
-        cqlParsingService.getCqlLookups(
-            lookupRequest.getCql(), lookupRequest.getMeasureExpressions(), accessToken));
+  public ResponseEntity<CqlBuilderLookup> getCqlBuilderLookups(
```
Comment (sb-cecilialiu) [inline src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlToolsController.java:97]: Would this affect UI since we change the return type?

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/model/CQLGraph.java [hunk 3]
```diff
@@ -49,19 +49,19 @@ public boolean isPath(String source, String destination) {
...
-        if (adjacentNode.equals(destination)) {
-          return true;
-        }
-
-        // if it's not the destination node and the node hasn't been visited yet, add it to the
-        // queue to be visited.
-        if (!visited.contains(adjacentNode)) {
-          visited.add(adjacentNode);
-          queue.add(adjacentNode);
+      if (this.graph.get(currentNode) != null) {
```
Comment (sb-cecilialiu) [inline src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/model/CQLGraph.java:52]: Nice work here, thanks a lot for being thorough!

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java [hunk 4]
```diff
@@ -556,18 +568,29 @@ private Element resolve(String identifier, CompiledLibrary library) {
           currentContext, def.getPath() + "-" + def.getVersion() + "|" + def.getLocalIdentifier());
       libraryAccessor = def;
       try {
-        parseChildLibraries(def);
-        libraries.add(
+        var parsedLibrary =
```
Comment (sb-cecilialiu) [inline src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java:571]: Nice work here!

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlConversionController.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlConversionController.java]: looks like something that I used for testing from UI.  good catch, thank @sb-cecilialiu!

File: src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlToolsController.java (hunk 2 above)
Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/controllers/CqlToolsController.java:97]: it should not. because this endpoint was QDM-specific which is no longer needed in FHIR service. So I replaced it with this one

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java (hunk 4 above)
Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/Cql2ElmListener.java:571]: thank you! noticed a few diffs between QDM translator and FHIR translator, so moved over them. This should be refactored to make it common logic between these two services.

File: src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/model/CQLGraph.java (hunk 3 above)
Comment (adongare) [reply src/main/java/gov/cms/mat/cql_elm_translation/utils/cql/parsing/model/CQLGraph.java:52]: thank you!

**Review by sb-cecilialiu — APPROVED** [review]
